### PR TITLE
Minimize or Close app after copying OTP #805

### PR DESCRIPTION
**Pull request by digitalheir**
Please give feedback

### CODE REVIEW

File: app/src/main/res/xml/preferences_behavior.xml [hunk 1]
```diff
@@ -8,6 +8,12 @@
         android:title="@string/pref_focus_search"
         android:summary="@string/pref_focus_search_summary"
         app:iconSpaceReserved="false"/>
+    <androidx.preference.SwitchPreferenceCompat
+        android:defaultValue="false"
+        android:key="pref_minimize_on_copy"
+        android:title="@string/pref_minimize_on_copy_title"
+        android:summary="Minimize app after copying token"
```
Comment (alexbakker) [inline app/src/main/res/xml/preferences_behavior.xml:15]: This should to be a translatable string as well. Also, slight modification suggestion: "Minimize the app after copying a token".

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -272,6 +272,7 @@
 
     <string name="pref_highlight_entry_title">Highlight tokens when tapped</string>
     <string name="pref_highlight_entry_summary">Make tokens easier to distinguish from each other by temporarily highlighting them when tapped</string>
+    <string name="pref_minimize_on_copy_title">Minimize app after copying token</string>
```
Comment (alexbakker) [inline app/src/main/res/values/strings.xml:275]: Let's make the title a bit shorter, and the summary a bit longer. For the title: "Minimize on copy".

File: app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java [hunk 3]
```diff
@@ -929,6 +932,9 @@ private void copyEntryCode(VaultEntry entry) {
         ClipboardManager clipboard = (ClipboardManager) getSystemService(Context.CLIPBOARD_SERVICE);
         ClipData clip = ClipData.newPlainText("text/plain", entry.getInfo().getOtp());
         clipboard.setPrimaryClip(clip);
+        if(_entryListView.isMinizeAfterCopyEnabled()) {
```
Comment (alexbakker) [inline app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java:935]: I don't think the EntryListView needs to know about this. We can simply check for ``_prefs.isMinimizeOnCopyEnabled()`` here.

Also, missing a space after ``if`` here.

File: app/src/main/res/values-ar-rSA/strings.xml [hunk 4]
```diff
@@ -184,6 +184,7 @@
     <string name="preference_manage_groups_summary">أدر واحذف مجموعاتك هنا</string>
```
Comment (alexbakker) [inline app/src/main/res/values-ar-rSA/strings.xml:184]: Please remove the translations to other languages. All translations are managed in Crowdin and the ones you've added here would be overwritten when we import them before the next release.

**Review by alexbakker — CHANGES_REQUESTED** [review]
Thanks for implementing this. I've left a couple of comments.

**Review by alexbakker — CHANGES_REQUESTED** [review]
Looks good. Please also remove the translations to other languages and squash your commits. After that, this'll be good to go!

**Comment by digitalheir** [comment]
I don't know how to squash commits

**Comment by alexbakker** [comment]
No problem. It's easy to learn though! There are lots of good articles about this online. Rebasing is also a good tool to have in your toolbox. As you've probably noticed, merging things back into your branch gets messy fairly quickly.

**Review by alexbakker — APPROVED** [review]
Thanks!

**Comment by juke32** [comment]
How about having it go to the last used app, the screen most likely to need the OTP after copy

**Comment by cacfd3a** [comment]
If you open Aegis while another app is on the foreground (like from a shortcut in the Quick Settings panel), it should already work like this.

In general, this is a responsibility of the operating system. I think apps cannot even query what is the most recently used app, because that is a pretty serious privacy and security risk.

**Comment by michaelschattgen** [comment]
>I think apps cannot even query what is the most recently used app

Correct. We are technically able to assign an app to each entry so that Aegis launches that app, but that will just launch a new intent. This will confuse users and they'll lose their current sign-in progress of said app.

**Pull request merged**